### ISSUE TITLE
Add caching for improved data performance

### ISSUE THREAD

**Issue by dougwaldron** [gaepdit/template-app#20]
See the [cache-aside pattern](https://devblogs.microsoft.com/dotnet/introducing-the-reliable-web-app-pattern/?mkt_tok=NDI2LVFWRC0xMTQAAAGK9hBNMINsxnhoeUiCxqjUC2A5SMmoMCp1FF-u0W06HaiJJoWyVYPwTVjGDAQdCzVgw-SjcL6MlnFykZ8UE-aJAPlsow7yEKlqCzD5AbYaRyt8DOa9#cache-aside-pattern) for an implementation suggestion.

**Comment by aaroniz-bgu** [comment]
Hi, I might be able to help.

Few questions before that, do you plan on having a distributed cache? one that would be share by multiple instances of this application? If so, note that it usually involves a third-party service, additionally I might not be able to help.

If not, what are your "hot-paths"? are you planning to chache all of your entity types, or just some of them?

I'll be waiting for your response.

**Comment by dougwaldron** [comment]
Hey @aaroniz-bgu , thanks for the offer to help!

Since this repo is intended to be used as a template for other projects, the goal is to implement basic features with examples that future developers can build on. So there is flexibility in how it could be implemented.

That said, we don't anticipate needing distributed caches, and different entities would be cached with different lifetimes or not at all. For the current example entities, you can assume that Offices are static over long periods of time, while Customers and Clients change more often. Customers are accessed frequently, and Clients infrequently. Let me know if that helps.

**Comment by dougwaldron** [comment]
@aaroniz-bgu I went ahead and assigned this issue to you, but no pressure of course! Let me know if you decide to work on it or change your mind. 😊

**Comment by aaroniz-bgu** [comment]
Sounds great, I'll start working on it today, probably later.
I'll take those notes into consideration.

**Comment by aaroniz-bgu** [comment]
So I have started, baby steps, firstly got familiar with the project. ([I've started with `Customer` just to make sure I'm on track](https://github.com/aaroniz-bgu/template-app-caching/tree/main)).

Not much to say, about it, I do have some questions if you'd like to comunicate over discord(_aaroniz) about them, really short, insignifact ones.

> That said, we don't anticipate needing distributed caches, and different entities would be cached with different lifetimes or not at all.

However, one question I would like to ask in this forum, is what entites you would most certainly wouldn't like to cache? I could guess which ones, but would like to know in advance.

Thanks!
